server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java

```diff
@@ -373,6 +378,59 @@ private void prepareAndValidatePublicationRequest(ProxyContext context, Publicat
         validateRules(publication);
     }
 
+    private static String buildTargetFolderForCustomAppFiles(Publication.Resource resource) {
+        String targetUrl = resource.getTargetUrl();
+        String appPath = targetUrl.substring(targetUrl.indexOf(ResourceDescriptor.PATH_SEPARATOR,
+                        targetUrl.indexOf(ResourceDescriptor.PATH_SEPARATOR) + 1) + 1,
+                targetUrl.lastIndexOf(ResourceDescriptor.PATH_SEPARATOR));
+        String appName = targetUrl.substring(targetUrl.lastIndexOf(ResourceDescriptor.PATH_SEPARATOR) + 1);
+        return appPath + ResourceDescriptor.PATH_SEPARATOR + "." + appName + ResourceDescriptor.PATH_SEPARATOR;
+    }
+
+    private void addCustomApplicationRelatedFiles(ProxyContext context, Publication publication) {
```

Comment:
The method is a good candidate for ApplicationSchemaService.

Reply:
We tried to move it in the previous comments but then desired to leave it as is there. Anyway, I suggest moving it away during the refactoring.